llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -6590,12 +6590,55 @@ bool CombinerHelper::matchRedundantBinOpInEquality(MachineInstr &MI,
   return CmpInst::isEquality(Pred) && Y.isValid();
 }
 
-bool CombinerHelper::matchShiftsTooBig(MachineInstr &MI) const {
+static std::optional<unsigned>
```

Comment:
This function needs a comment. It seems like it is returning the min useless shift amount, not the max useful one.

Reply:
True that the name is wrong, updated.